### ISSUE TITLE
No valid mission available error after reboot w/o sending new flight plan

### ISSUE THREAD

**Issue by jgoppert** [PX4/Firmware#2564]
It appears that the mission is not checked and the valid status isn't set until the user manually re-sends the flight plan after reboot.

**Comment by jgoppert** [comment]
I've tracked it down to this if statement where the checker is run. Is it possible that mission type is not none when a valid flight plan is still in memory but has not been validated yet?

```c++
/* check if the home position became valid in the meantime */
		if ((_mission_type == MISSION_TYPE_NONE || _mission_type == MISSION_TYPE_OFFBOARD) &&
			!_home_inited && _navigator->home_position_valid()) {
```

**Comment by LorenzMeier** [comment]
Not sure - it should be MISSION_TYPE_OFFBOARD since that is where the mission came from.

**Comment by LorenzMeier** [comment]
Any new info on this one? I believe its real and I have other reports supporting the findings here.

**Comment by jgoppert** [comment]
No new insights. But we confirmed it with another 10 flights or so.

**Comment by LorenzMeier** [comment]
Fixed in 253b8f5. Let me know if I missed any corner cases.
